components/remote-environment-broker/internal/broker/middleware.go

```diff
@@ -4,16 +4,48 @@ import (
 	"net/http"
 
 	osb "github.com/pmorie/go-open-service-broker-client/v2"
+	"github.com/sirupsen/logrus"
 )
 
 // OSBContextMiddleware implements Handler interface
-type OSBContextMiddleware struct{}
+type OSBContextMiddleware struct {
+	brokerFlavor brokerFlavorProvider
+	log          *logrus.Entry
+}
+
+//go:generate mockery -name=brokerFlavorProvider -output=automock -outpkg=automock -case=underscore
+type brokerFlavorProvider interface {
+	IsClusterScoped() bool
+	GetNsFromBrokerURL(url string) (string, error)
+}
+
+// NewOsbContextMiddleware created OsbContext middleware
+func NewOsbContextMiddleware(brokerFlavorProvider brokerFlavorProvider, log *logrus.Entry) *OSBContextMiddleware {
+	return &OSBContextMiddleware{
+		brokerFlavor: brokerFlavorProvider,
+		log:          log.WithField("service", "OSBContextMiddleware"),
+	}
+}
 
 // ServeHTTP adds content of Open Service Broker Api headers to the requests
-func (OSBContextMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+func (m *OSBContextMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	brokerNamespace := ""
+	if m.brokerFlavor.IsClusterScoped() == false {
+		var err error
+		brokerNamespace, err = m.brokerFlavor.GetNsFromBrokerURL(r.Host)
+		if err != nil {
+			errMsg := "misconfiguration, broker is running as a namespace-scoped, but cannot extract namespace from request host"
```

Comment:
"(..) from request Host header"

Reply:
from request.go:
```
// For server requests Host specifies the host on which the
	// URL is sought. Per RFC 2616, this is either the value of
	// the "Host" header or the host name given in the URL itself.
	// It may be of the form "host:port". For international domain
	// names, Host may be in Punycode or Unicode form. Use
	// golang.org/x/net/idna to convert it to either format if
	// needed.
	//
	// For client requests Host optionally overrides the Host
	// header to send. If empty, the Request.Write method uses
	// the value of URL.Host. Host may contain an international
	// domain name.
	Host string
```
so it will be safer to write "from request"